### ISSUE TITLE
Outputted netCDF files not readable by ncdump, ncopen command line tools

### ISSUE THREAD

**Issue by spencerahill** [spencerahill/aospy#58]
For example:

```
ncdump -h /work/s1h/aero_3agcm/am2/reyoi_cont/t_surf/t_surf.jja.std.from_monthly_ts.am2.reyoi_cont.1983-1984.nc
*** ncdump: ncopen failed on /work/s1h/aero_3agcm/am2/reyoi_cont/t_surf/t_surf.jja.std.from_monthly_ts.am2.reyoi_cont.1983-1984.nc
```

The official UCAR documentation is hard to read (as in a couple minutes of googling didn't get me anywhere), so it's not clear to me what exactly our files are missing or doing wrong.  Interesting that those command line tools don't work whereas netCDF4-python and xarray do.

I am labeling this a bug, because I assumed previously that this would work, and ultimately it should.  We should add some tests for this; for starters we could just the example file I used here.

**Comment by spencerkclark** [comment]
I'm actually able to run `ncdump -h` on this file without error.  What platform are you trying to run this on?

**Comment by spencerahill** [comment]
Good catch.  The ncdump in my Anaconda distribution isn't working on any netCDF files:

```
$ which ncdump
/home/s1h/anaconda/bin/ncdump
$ ncdump -h /archive/s1h/am2/am2clim_reyoi/gfdl.ncrc2-default-prod/pp/atmos/atmos.static.nc
*** ncdump: ncopen failed on /archive/s1h/am2/am2clim_reyoi/gfdl.ncrc2-default-prod/pp/atmos/atmos.static.nc
$ /usr/local/bin/ncdump -h /archive/s1h/am2/am2clim_reyoi/gfdl.ncrc2-default-prod/pp/atmos/atmos.static.nc
netcdf atmos.static { // format variant: 64bit
dimensions:
    lat = 90 ;
    lon = 144 ;
    phalf = 25 ;
    latb = 91 ;
    lonb = 145 ;
...
```

/usr/local/bin/ncdump also works on aospy-outputted files.  Very weird -- I use ncdump pretty often, and this never occurred before.  But not an aospy problem, so closing.

**Comment by spencerahill** [comment]
FYI uninstalling and then reinstalling libnetcdf via conda solved this

**Comment by meyerbe** [comment]
Hi 
I have the same problem since I installed the Anaconda package: ncdump does not work. I tried as you suggested to uninstall and reinstall libnetcdf, but it did not solve it. By any chance, do you have experience what I could do?

**Comment by spencerahill** [comment]
Hi @meyerbe, I am actually having this same problem on my new macbook running MacOS with libnetcdf 4.4.1-0 installed via conda.  But the version in my /usr/local/bin that I got from homebrew works fine.  

So if you're on a Mac, then downloading it via homebrew (and putting `/usr/local/bin` before your conda bin directory in your PATH) is an easy workaround.

What OS are you on, and what are your conda and libnetcdf versions?

**Comment by spencerkclark** [comment]
@meyerbe @spencerahill it seems [there may be an issue with libnetcdf in the default conda channel](https://groups.google.com/a/continuum.io/forum/m/#!topic/anaconda/gduHk0_b5o0).  Have either of you tried uninstalling hdf4, hdf5, libnetcdf, and netcdf4 from your conda environments, and reinstalling them from the conda-forge channel?
```
$ conda remove hdf4 hdf5 libnetcdf netcdf4
Fetching package metadata .......
Solving package specifications: ..........

Package plan for package removal in environment /path/to/test_environment:

The following packages will be REMOVED:

    hdf4:      4.2.11-0
    hdf5:      1.8.17-1
    libnetcdf: 4.4.1-0
    netcdf4:   1.2.4-np111py27_1

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ]|#################################################| 100%

$ conda install -c conda-forge netcdf4
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment /path/to/test_environment:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    hdf4-4.2.12                |                0         920 KB  conda-forge
    hdf5-1.8.17                |                8         4.0 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         4.9 MB

The following NEW packages will be INSTALLED:

    hdf4:        4.2.12-0          conda-forge
    hdf5:        1.8.17-8          conda-forge
    libgfortran: 3.0.0-0           conda-forge
    libnetcdf:   4.4.1.1-2         conda-forge
    netcdf4:     1.2.6-np111py27_0 conda-forge

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|#################################################| 100%
```

I'm able to reproduce your issues in a temporary environment (with netcdf4 installed via the default channel) -- the two shell commands above fix things in that environment; in my working environment I have those libraries installed via the conda-forge channel and ```ncdump``` works great.

**Comment by spencerahill** [comment]
@spencerkclark thanks!  `ncdump` works for me with these conda-forge versions of the libraries.

@meyerbe does that solve your problem?

**Comment by meyerbe** [comment]
Great, that also works for me! Thanks a lot! 
@spencerkclark: do you have an explanation for it? Is it a problem of some package version that is installed in the default conda installation?

**Comment by spencerkclark** [comment]
@meyerbe @spencerahill great, I'm glad we found a solution!  

I wish I could offer a good explanation; however I'm not very familiar with the intricacies of installing the compiled (i.e. non-python) parts of the netCDF ecosystem.  The author of the first email in the thread I linked to may be on to something regarding the differences in "linking" between the ```ncdump``` and various compiled libraries between the default and conda-forge installations, but I'm not knowledgeable enough to say for sure.
